Client/Frontend/Browser/Playlist/Cells/PlaylistCell.swift

```diff
@@ -0,0 +1,208 @@
+            $0.bottom.lessThanOrEqualTo(iconStackView.snp.bottom)
+        }
+        
+        separator.snp.makeConstraints {
+            $0.left.equalTo(titleLabel.snp.left)
+            $0.right.bottom.equalToSuperview()
+            $0.height.equalTo(1.0 / UIScreen.main.scale)
+        }
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func updateThumbnail() {
+        if let image = thumbnailView.image {
+            let boundsScale = thumbnailView.bounds.size.width / thumbnailView.bounds.size.height
+            let imageScale = image.size.width / image.size.height
+            var drawingRect = thumbnailView.bounds
+
+            if boundsScale > imageScale {
+                drawingRect.size.width =  drawingRect.size.height * imageScale
+                drawingRect.origin.x = (thumbnailView.bounds.size.width - drawingRect.size.width) / 2.0
+            } else {
+                drawingRect.size.height = drawingRect.size.width / imageScale
+                drawingRect.origin.y = (thumbnailView.bounds.size.height - drawingRect.size.height) / 2.0
+            }
+
+            let path = UIBezierPath(roundedRect: drawingRect, cornerRadius: 5.0)
+            thumbnailMaskView.path = path.cgPath
+            thumbnailView.layer.mask = thumbnailMaskView
+        }
+    }
+    
+    override var layoutMargins: UIEdgeInsets {
+        get {
+            return .zero
+        }
+
+        set (newValue) {
+            _ = newValue
+            super.layoutMargins = .zero
+        }
+    }
+    
+    override var separatorInset: UIEdgeInsets {
+        get {
+            return UIEdgeInsets(top: 0, left: self.titleLabel.frame.origin.x, bottom: 0, right: 0)
+        }
+        
+        set (newValue) {
+            _ = newValue
+            super.separatorInset = UIEdgeInsets(top: 0, left: self.titleLabel.frame.origin.x, bottom: 0, right: 0)
+        }
+    }
+    
+    // MARK: - Thumbnail
+    
+    func loadThumbnail(item: PlaylistInfo, onDurationUpdated: ((TimeInterval?) -> Void)? = nil) {
+        guard let url = URL(string: item.src) else { return }
+        
+        // Loading from Cache failed, attempt to fetch HLS thumbnail
+        self.thumbnailGenerator = HLSThumbnailGenerator(url: url, time: 3, completion: { [weak self] image, trackDuration in
+            guard let self = self else { return }
+            
+            if let trackDuration = trackDuration {
+                onDurationUpdated?(trackDuration)
+            }
+            
+            if let image = image {
+                self.thumbnailImage = image
+                self.thumbnailGenerator = nil
+                SDImageCache.shared.store(image, forKey: url.absoluteString, completion: nil)
+            } else {
+                // We can fall back to AVAssetImageGenerator or FavIcon
+                self.loadThumbnailFallbackImage(item: item)
+            }
+        })
+    }
```

Comment:
This logic doesn't belong inside the cell

Reply:
Same logic as `LargeFaviconView` where the owner has to be the view. The cell owns the:

```swift
private var favIconFetcher: FaviconFetcher?
private var thumbnailGenerator: HLSThumbnailGenerator?
```

So when the cell is recycled and the image is loaded in `cellForRowAtIndexPath`, it will cancel the loading of favIcon and thumbnails for that cell.

I made the variables accessible from the controller and I moved the logic there.